### PR TITLE
Add ALAC 24bit

### LINKED ISSUE #621: Add Alac 24Bits in profile
Hello

Many thanks for the good work with this app.
I find it really helpfull to organize and maintain a music library with the best possible audio tracks quality !!

For itunes compatibility reasosn, i have converted all my flac24 files to alac 24 bits to have all my music both on my audiophile music player (volumio on a RPI + dac) and my macbook. I was fed up having duplicates between my alac classicla lossless 16 bits and what i found with flac24bits

Now that my organisational issue is solved with all my 24 bits files integrated to itunes thx to alac24, i m lost with lidarr that does not seem to consider ala24 as a hightest audio quality format than standard lossless file.
Could it be possible to create an alac 24bit profile (like what have been done with flac 24) to solve this situation ?

Many thx !!

Best Regards
Al

[AB#323](https://dev.azure.com/Servarr/7ab38f4e-5a57-4d70-84f4-94dd9bc5d6df/_workitems/edit/323)

### PR DESCRIPTION
**Pull request by jeffbyrnes**
#### Database Migration
Yes - 072

#### Description

Add an ALAC 24bit quality

#### Todos

- [ ] Tests
- [x] Translation Keys (./src/NzbDrone.Core/Localization/Core/en.json)
- [ ] [Wiki Updates](https://wiki.servarr.com)

#### Issues Fixed or Closed by this PR

* Fixes #621

### CODE REVIEW

File: src/NzbDrone.Core/Datastore/Migration/004_add_various_qualities_in_profile.cs [hunk 1]
```diff
@@ -134,6 +135,7 @@ public enum Qualities4
         VORBIS_Q5,
         WMA,
         FLAC_24,
+        ALAC_24,
```
[the remark this answers is not in the document]
Comment (Qstick) [reply src/NzbDrone.Core/Datastore/Migration/004_add_various_qualities_in_profile.cs]: Agree.. As well we shouldn't be making changes to this file. Qualities4 means this was the state of qualities at migration 4. Modifying old migrations isn't usually a great practice.

**Review by Qstick — CHANGES_REQUESTED** [review]
One additional comment. This also needs a new migration

File: src/NzbDrone.Core/Datastore/Migration/004_add_various_qualities_in_profile.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (jeffbyrnes) [reply src/NzbDrone.Core/Datastore/Migration/004_add_various_qualities_in_profile.cs]: Fixed in 36f1cb1315

File: src/NzbDrone.Core/Parser/QualityParser.cs [hunk 2]
```diff
@@ -133,7 +133,15 @@ public static QualityModel ParseQuality(string name, string desc, int fileBitrat
...
                 case Codec.ALAC:
-                    result.Quality = Quality.ALAC;
+                    if (sampleSize == SampleSize.S24)
+                    {
+                        result.Quality = Quality.ALAC_24;
+                    {
+                    else
+                    {
+                        result.Quality = Quality.ALAC;
+                    }
```
[the remark this answers is not in the document]
Comment (jeffbyrnes) [reply src/NzbDrone.Core/Parser/QualityParser.cs:143]: Definitely appreciate this suggestion, makes sense to me!

That said, just above is this snippet, and I was aiming to be consistent in style:

```c#
case Codec.FLAC:
    if (sampleSize == SampleSize.S24)
    {
        result.Quality = Quality.FLAC_24;
    }
    else
    {
        result.Quality = Quality.FLAC;
    }
```

File: src/NzbDrone.Core/Datastore/Migration/004_add_various_qualities_in_profile.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (jeffbyrnes) [reply src/NzbDrone.Core/Datastore/Migration/004_add_various_qualities_in_profile.cs]: @Qstick gotcha; I didn’t catch this is a migration! I’ll rework this. First time trying to make tweaks, and doing so blindly as someone who’s never worked with C# 🙂

File: src/NzbDrone.Core/Parser/QualityParser.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (Qstick) [reply src/NzbDrone.Core/Parser/QualityParser.cs:143]: Yes, let's change both. Thanks!

**Comment by Qstick** [comment]
Migration 19 is a good example of what you'll need to do with the new migration.

**Comment by jeffbyrnes** [comment]
Thanks @Qstick, I’ll use that as the basis for my new migration!

**Comment by jeffbyrnes** [comment]
@Qstick, @mynameisbogdan, ok so I’m still flailing around pretty blindly, again, zero C# experience, but I’m familiar with code-based DB migrations (longtime Rubyist here, mostly used them via Ruby on Rails), so maybe ae478d3 is close?

**Comment by jeffbyrnes** [comment]
Looks like things changed since migration 19, since this error is showing up all over: `no such table: Profiles`

I’ll poke around som more.

**Comment by jeffbyrnes** [comment]
> Should be QualityProfiles. 

Ah! Thanks.

**Comment by jeffbyrnes** [comment]
Looks like the build just timed out.

**Comment by jeffbyrnes** [comment]
Sorry y’all, I’m not familiar enough to figure out what’s going on. Trying a build locally to see if that tells me more, but again, not a C# dev so very much stumbling around in the dark 😆

**Comment by Qstick** [comment]
It's ok, stumbling in the dark is how we all learned c#. Take you time and feel free to ask questions here or discord

**Comment by jeffbyrnes** [comment]
Hey thanks @mynameisbogdan, looks like all testing is passing with your help. I’ll look over what you changed compared to what I had so I can learn what I missed/did incorrectly.

**Comment by jeffbyrnes** [comment]
Thanks @mynameisbogdan

**Comment by jeffbyrnes** [comment]
Woohoo! Thanks for the help here y'all.

When will a new Docker image be baked that includes this?

**Pull request merged**